MigrationScript/AmsOperationsHelper.ps1

```diff
@@ -0,0 +1,294 @@
+    }
+    catch
+    {
+        $putErrorMsg = $_.ErrorDetails | ConvertTo-Json -Depth 10;
+		Write-Host ($($putErrorMsg.error.code));
+		$logger.LogError("Put-AmsV2Provider : Failed with error: ($($putErrorMsg.error.code))", "", "");
+    }
+
+	return @{
+		Response = $response
+		Error = $putErrorMsg
+	}
+}
+
+<#
+.SYNOPSIS
+Function to get the deployment status for Provider.
+
+.PARAMETER subscriptionId
+SubscriptionId of the AMS v2 Monitor.
+
+.PARAMETER resourceGroup
+Resource Group Name of the AMS v2 Monitor.
+
+.PARAMETER monitorName
+AMS v2 Monitor Name.
+
+.PARAMETER providerName
+Provider name.
+
+.PARAMETER logger
+logger object. 
+
+.EXAMPLE
+$logger = New-Object ConsoleLogger
+GetAmsV2ProviderStatus -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -providerName $providerName -logger $logger;
+#>
+function GetAmsV2ProviderStatus([string]$subscriptionId, [string]$resourceGroup, [string]$monitorName, [string]$providerName, $logger)
+{
+    $rawToken = Get-AzAccessToken -ResourceTypeName Arm
+    $armToken = $rawToken.Token
+	$apiVersion = "2021-12-01-preview";
+
+    $headers = @{
+        "Content-Type" = "application/json"
+        "Authorization" = "Bearer $armToken"
+    }
+
+    [string]$url = "https://management.azure.com/";
+	[string]$subscriptionParams = "subscriptions/" + $subscriptionId;
+	[string]$rgParams = "/resourceGroups/" + $resourceGroup;
+	[string]$providerParams = "/providers/Microsoft.Workloads/monitors/" + $monitorName + "/providerInstances/" + $providerName + "?api-version=" + $apiVersion;
+	$url = $url + $subscriptionParams + $rgParams + $providerParams;
+	[string]$provisiongState = "";
+	$logger.LogInfo("Making Get Provider call with $url")
+    
+	try
+    {
+        $response = Invoke-RestMethod -Method 'get' -Uri $url -Headers $headers;
+		$provisiongState = $response.properties.provisioningState
+    }
+    catch
+    {
+        $GetProviderErrorMsg = $_.ErrorDetails.ToString();
+		$logger.LogError("GetAmsV2ProviderStatus : Failed with error: ($($GetProviderErrorMsg)))", "500", "");
+		$provisiongState = "Not Created"
+    }
+
+	return @{
+		Response = $response
+		provisiongState = $provisiongState
+		Error = $GetProviderErrorMsg
+	}
+}
+
+<#
+.SYNOPSIS
+Function to get the AMS v2 Monitor Properties.
+
+.PARAMETER subscriptionId
+SubscriptionId of the AMS v2 Monitor.
+
+.PARAMETER resourceGroup
+Resource Group Name of the AMS v2 Monitor.
+
+.PARAMETER monitorName
+AMS v2 Monitor Name.
+
+.PARAMETER logger
+logger object. 
+
+.EXAMPLE
+$logger = New-Object ConsoleLogger
+GetAmsV2MonitorProperties -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -logger $logger
+#>
+function GetAmsV2MonitorProperties([string]$subscriptionId, [string]$resourceGroup, [string]$monitorName, $logger)
+{
+    $rawToken = Get-AzAccessToken -ResourceTypeName Arm
+    $armToken = $rawToken.Token
+	$apiVersion = "2021-12-01-preview";
```

Comment:
Lets have a constants file and use those here :) If thats possible to do in PowerShell that is